### PR TITLE
Loki: Pass time range variable in variable editor

### PR DESCRIPTION
**Pull request by ivanahuckova**
Discovered while I was investigating https://github.com/grafana/grafana/issues/80508, but this is not relevant to the issue, because it affects only showing of label values in editor. But it linked to same fix in Prometheus https://github.com/grafana/grafana/pull/78950 so I created this fix in here as well.

### CODE REVIEW

File: public/app/plugins/datasource/loki/components/VariableQueryEditor.test.tsx [hunk 1]
```diff
@@ -133,4 +134,27 @@ describe('LokiVariableQueryEditor', () => {
     await select(screen.getByLabelText('Label'), 'luna', { container: document.body });
     await screen.findByText('luna');
   });
+
+  test('Calls language provider fetchLabels with the time range received in props', async () => {
```
Comment (matyax) [inline public/app/plugins/datasource/loki/components/VariableQueryEditor.test.tsx:138]: <3

**Review by matyax — APPROVED** [review]

**Comment by ivanahuckova** [comment]
@matyax I have noticed that when we started to pass `range` and added it to `useEffect []` it was triggering a lot of requests. Therefore I have updated a logic to only run `fetchLabel` if the type has changed to `LabelValues`. I have also added additional tests. Could you please re-review. Thanks!

**Comment by matyax** [comment]
Good catch! Even with these changes, I see that it makes 3 requests for every change, not sure why.

https://github.com/grafana/grafana/assets/1069378/b2fb63ae-c7a0-4ead-aebc-25ea36513a81

**Comment by ivanahuckova** [comment]
@matyax did you `git pull` ? Cause I can't reproduce: 

https://github.com/grafana/grafana/assets/30407135/6b8140e9-2e7d-411a-acca-bb71061c0040

**Comment by matyax** [comment]
I'll test again. I pulled but maybe didn't wait for the build?

**Review by matyax — APPROVED** [review]
Perfect!

**Pull request merged**